gateway/hostname.go

```diff
@@ -48,6 +48,13 @@ func NewHostnameHandler(c Config, backend IPFSBackend, next http.Handler) http.H
 			host = xHost
 		}
 
+		// Comply with user agents that explicitly requested plain path processing
+		// (used by CLI and non-browser tools to disable Host-based subdomain redirects etc)
+		if mode := r.Header.Get("Ipfs-Gateway-Mode"); mode == "path" {
```

Comment:
It would be _nice_ if we could comply with "domain" as well and/or _reject_ the request if we can't. But that's not all that critical, TBH.

Reply:
@Stebalien  you mean the `Ipfs-Gateway-Mode`  header should  be respected only if the domain name present in `Host` HTTP header is preconfigured  in [Gateway.PublicGateways](https://github.com/ipfs/kubo/blob/master/docs/config.md#gatewaypublicgateways), or something else?

Reply:
I mean: I'd expect it to _also_ check some form of `mode == "domain"` and, in that case, refuse to act as a "path" gateway. Of course, I don't see how that would be all that _useful_...

Reply:
Hm.. we could add support for values other than `path`  in the future, but no clear use case atm.

Frankly, anything other than "downgrading" from `subdomain` to `path`, like "upgrading" to subdomain or dnslink hosting with proper Origin isolation requires custom setup and usually will not be possible out of the box:
-  `127.0.0.1:8080` is unable to provide subdomain isolation even when you ask for it. (That is why we use `localhost:8080` for local subsomain)
- non-localhost dnslink and subdomain gateways require a valid **wildcart** TLS cert

Reply:
Yeah, you're right.